src/modules/post/graphql/mutation/post_create.rs

```diff
@@ -28,9 +28,16 @@ impl PostCreate {
         let token = auth.token()?;
         let user = services.auth.whoami(token).await?;
 
-        match services.post.create(user, input).await {
+        match services.post.create(user.clone(), input).await {
```

Comment:
This needs refactor, instead of cloning I would like to be able to borrow the `User` instance here. But given that this PR is focused on providing `DataLoader` support I'm leaving it for a future PR. @Dave136 feel free to work on this if you want to!